### PR TITLE
Fix #1294: docs for AWS EC2 authentication

### LINKED ISSUE #1294: AWS KMS - support authentication from instance metadata of EC2
**Is your feature request related to a problem? Please describe.**

This issue separates out one of the two enhancement requests described by https://github.com/kroxylicious/kroxylicious/issues/1282.

Currently the proxy requires a long-lived AWS access key/secret key combination to authenticate to AWS. Whilst this
authentication is supported, as it not seen as the most secure option.

For use cases when the proxy binary is running directly on Amazon EC2,  the preferred option is for the EC2 instance to be associated with an IAM role, then the application [fetch short lived credentials from the instance metadata](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials).  

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Documentation

### Description

Docs associated for #1294 that show the user how to configure record encryption to authenticate via AWS EC2 metadata. 

This PR will remain draft until such time that #1442 (the PR delivering the feature) gets accepted. 

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Update documentation
- [ ] Make sure all unit/integration tests pass
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] If applicable to the change, [trigger the system test suite](../blob/main/DEV_GUIDE.md#jenkins-pipeline-for-system-tests).  Make sure tests pass.
- [ ] If applicable to the change, [trigger the performance test suite](../blob/main/PERFORMANCE.md#jenkins-pipeline-for-performance). Ensure that any degradations to performance numbers are understood and justified.
- [ ] Ensure the PR references relevant issue(s) so they are closed on merging.
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

> **_NOTE:_**  You must be a member of `@kroxylicious/developers` to trigger the system test and performance test suites.  If you are not part of this group, comment on the PR requesting a trigger, tagging `@kroxylicious/developers`.

### CODE REVIEW

File: docs/modules/record-encryption/aws-kms/con-aws-kms-setup.adoc [hunk 1]
```diff
@@ -90,13 +71,11 @@ EOF
 aws iam create-policy --policy-name KroxyliciousRecordEncryption --policy-document file:///tmp/policy
 ----
 
-== Apply the alias-based policy to the filter's application identity
+== Establish an authentication mechanism for the filter
+
+The filter must authenticate to AWS in order to perform envelope encryption operations, such as generating and decrypting DEKs.
+
+include::con-aws-kms-setup-application-identity-iam.adoc[leveloffset=+1]
+include::con-aws-kms-setup-application-identity-ec2-metadata.adoc[leveloffset=+1]
```
Comment (PaulRMellor) [inline docs/modules/record-encryption/aws-kms/con-aws-kms-setup.adoc:79]: We could conditionalize to leave out of downstream here too

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+
+= Authenticating using a EC2 metadata
+
+[source, yaml]
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc:4]: I thought we could change from subsection to title here to keep the content in the same section

**Review by PaulRMellor — COMMENTED** [review]
Thanks Keith. I left a few minor comments around wording and suggested we use condition for the upstream-only content.  

Outside this PR, I also think we could make the authentication steps into procedures to be more concise

**Comment by k-wall** [comment]
@PaulRMellor how does that look now?

**Comment by PaulRMellor** [comment]
> @PaulRMellor how does that look now?

Hi @k-wall  - Were the collapsed comments missed?

**Comment by k-wall** [comment]
@PaulRMellor please look again, I believe I have them all now.

File: docs/modules/record-encryption/aws-kms/con-aws-kms-setup.adoc (hunk 1 above)
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-setup.adoc]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-setup-application-identity-iam.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[id='con-aws-kms-setup-application-identity-iam-{context}']
+
+= Using an AWS IAM identity
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-setup-application-identity-iam.adoc]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+----
+kms: AwsKmsService                                            # <1>
+kmsConfig:
+  endpointUrl: https://kms.<region>.amazonaws.com             # <2>
+  credentialsProvider:
+    type: ec2Metadata
+    iamRole: KroxyliciousInstance                             # <3>
+  region: <region>                                            # <4>
+----
+<1> Name of the KMS provider. This must be `AwsKmsService`.
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<2> AWS Endpoint URL.  This must include the `https://` scheme part.
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc:7]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+
+= Authenticating using a EC2 metadata
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc]: done

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config.adoc [hunk 7]
```diff
@@ -7,24 +7,8 @@
...
-  region: <region>                                            # <6>
-----
-<1> Name of the KMS provider. This must be `AwsKmsService`.
-<2> AWS Endpoint URL.  This must include the `https://` scheme part.
-<3> (Optional) TLS trust configuration.
-<4> File containing the AWS Access Key
-<5> File containing the AWS Secret Key
-<6> AWS region identifier (e.g. `us-east-1`)
+include::con-aws-kms-service-config-identity-iam.adoc[leveloffset=+1]
+include::con-aws-kms-service-config-identity-ec2-metadata.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config.adoc]: done

**Review by robobario — APPROVED** [review]
LGTM with nits

File: docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+kmsConfig:
+  endpointUrl: https://kms.<region>.amazonaws.com             # <2>
+  ec2MetadataCredentials:
+    iamRole: <name of IAM role>                               # <3>
+    metadataEndpoint: <EC2 metadata endpoint>                 # <4>
+    credentialLifetimeFactor: 0.8                             # <5>
+  region: <region>                                            # <6>
+----
+<1> Specifies the name of the KMS provider. Use `AwsKmsService`.
+<2> AWS KMS endpoint URL,  which must include the `https://` scheme.
```
[the remark this answers is not in the document]
Comment (k-wall) [reply docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc:19]: Fair point. As this is a comment against existing comment, let's treat it as a separate issue.

**Pull request merged**